soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp

```diff
@@ -0,0 +1,1046 @@
+    if (doDraw)
+        ImGui::TreePop();
+
+    ImGui::EndTable(); //Checks Lead-out
+    ImGui::EndTable(); //Quick Options Lead-out
+    EndFloatWindows();
+    if (doingCollapseOrExpand) {
+        optCollapseAll = false;
+        optExpandAll = false;
+    }
+}
+
+// Windowing stuff
+void BeginFloatWindows(std::string UniqueName, ImGuiWindowFlags flags) {
+    ImGuiWindowFlags windowFlags = flags;
+
+    if (windowFlags == 0) {
+        windowFlags |=
+            ImGuiWindowFlags_AlwaysVerticalScrollbar | ImGuiWindowFlags_NoFocusOnAppearing;
+    }
+
+    if (!CVar_GetS32("gCheckTrackerWindowType", 1)) {
+        ImGui::SetNextWindowViewport(ImGui::GetMainViewport()->ID);
+        windowFlags |= ImGuiWindowFlags_NoDocking | ImGuiWindowFlags_NoNav | ImGuiWindowFlags_NoTitleBar |
+                       ImGuiWindowFlags_NoScrollWithMouse | ImGuiWindowFlags_NoScrollbar;
+
+        if (!CVar_GetS32("gCheckTrackerHudEditMode", 0)) {
+            windowFlags |= ImGuiWindowFlags_NoInputs | ImGuiWindowFlags_NoMove;
+        }
+    }
+    ImGui::PushStyleColor(ImGuiCol_WindowBg, Color_Background);
+    ImGui::PushStyleColor(ImGuiCol_Border, ImVec4(0, 0, 0, 0));
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowRounding, 4.0f);
+    ImGui::Begin(UniqueName.c_str(), nullptr, windowFlags);
+}
+void EndFloatWindows() {
+    ImGui::PopStyleVar();
+    ImGui::PopStyleColor();
+    ImGui::PopStyleColor();
+    ImGui::End();
+}
+
+uint8_t gRandomizeShopsanity;
+uint8_t gRandomizeShuffleTokens;
+uint8_t gRandomizeShuffleBeans;
+uint8_t gRandomizeShuffleScrubs;
+uint8_t gRandomizeShuffleMerchants;
+uint8_t gRandomizeShuffleCows;
+uint8_t gRandomizeShuffleAdultTrade;
+uint8_t gRandomizeShuffleKokiriSword;
+uint8_t gRandomizeShuffleWeirdEgg;
+uint8_t gRandomizeShuffleGerudoCard;
+uint8_t gRandomizeShuffleFrogSongRupees;
+uint8_t gRandomizeStartingMapsCompasses;
+uint8_t gRandomizeKeysanity;
+uint8_t gRandomizeGerudoFortressKeys;
+uint8_t gRandomizeBossKeysanity;
+uint8_t gRandomizeShuffleGanonBossKey;
+uint8_t gRandomizeGerudoFortress;
+
+void SetVanillaSettings() {
+    gRandomizeShopsanity = RO_SHOPSANITY_OFF;
+    gRandomizeShuffleTokens = RO_TOKENSANITY_ALL;
+    gRandomizeShuffleBeans = RO_GENERIC_YES;
+    gRandomizeShuffleScrubs = RO_GENERIC_YES;
+    gRandomizeShuffleMerchants = RO_GENERIC_YES;
+    gRandomizeShuffleCows = RO_GENERIC_NO;
+    gRandomizeShuffleAdultTrade = RO_GENERIC_YES;
+    gRandomizeShuffleKokiriSword = RO_GENERIC_YES;
+    gRandomizeShuffleWeirdEgg = RO_GENERIC_YES;
+    gRandomizeShuffleGerudoCard = RO_GENERIC_YES;
+    gRandomizeShuffleFrogSongRupees = RO_GENERIC_NO;
+    gRandomizeStartingMapsCompasses = RO_DUNGEON_ITEM_LOC_ANYWHERE;
+    gRandomizeKeysanity = RO_DUNGEON_ITEM_LOC_ANYWHERE;
+    gRandomizeGerudoFortressKeys = RO_GERUDO_KEYS_VANILLA;
+    gRandomizeBossKeysanity = RO_DUNGEON_ITEM_LOC_ANYWHERE;
+    gRandomizeShuffleGanonBossKey = RO_GANON_BOSS_KEY_ANYWHERE;
+    gRandomizeGerudoFortress = RO_GF_NORMAL;
+}
```

Comment:
this just feels strange (in vanilla beans aren't shuffled), i know the logic of `IsVisibleInCheckTracker` relies on that, but i think it'd actually be cleaner to just have 2 blocks in there depending on if rando is enabled or not

Reply:
Cleaned up this with 69719abf7054f0d0fba3a3b8c55f94f414c9db7e